pkg/daemon/daemon.go

```diff
@@ -177,20 +184,21 @@ configRetry:
 			continue configRetry
 		}
 
-		// We should now have enough config to connect to the datastore.
-		datastoreConfig := configParams.DatastoreConfig()
-
-		t.DatastoreClient, err = t.NewClientV3(datastoreConfig)
+		// Validate the config params
+		err = configParams.Validate()
```

Comment:
Why has this validate check moved from after merging datastore config, to before?

Reply:
I realised that doing it after didn't make sense for Typha, which doesn't load its config from the datastore.